### PR TITLE
spm: add upgrade checklist for SPM

### PR DESCRIPTION
**Pull request by Reminiscent**
### First-time contributors' checklist 

- [X] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add upgrade checklist for SPM.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
Related to https://github.com/pingcap/tidb/issues/27185
https://github.com/pingcap/tidb/issues/27064
https://github.com/pingcap/tidb/pull/26333.

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@Reminiscent 请@一位或两位 tech reviewer，谢谢。

**Comment by Reminiscent** [comment]
@eurekaka @rebelice PTAL

File: sql-plan-management.md [hunk 1]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+  select bind_sql from mysql.bind_info where source != 'builtin' and status = 'using';
+  
+  -- 将上一条查询得到的结果，在新版本的测试环境中进行验证。
+  
+  bind_sql_0;
+  bind_sql_1;
+  ...
+  
+  -- 如果报错信息是语法错误（ERROR 1064 (42000): You have an error in your SQL syntax），则需要删除对应的绑定。
+  -- 如果是其他错误，如未找到表，则表示语法兼容，不需要进行额外的处理。
```
Comment (TomShawn) [inline sql-plan-management.md:359]: 这两条注释建议放到代码块外。

**Review by eurekaka — APPROVED** [review]

File: sql-plan-management.md [hunk 2]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
     | max_execution_time | 查询过程最多消耗多少时间 |
 
 + `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读。由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
+
+## 升级检查 (Upgrade Checklist)
+
+执行计划管理功能（SPM）在版本升级过程中可能会出现一些兼容性问题导致升级失败，你需要在版本升级前做一些检查，确保版本顺利升级。
+
+* 当你尝试从 v5.2 以前的版本升级到此版本，需要注意在升级前检查自动演进的开关 `tidb_evolve_plan_baselines` 是否已经关闭，如果尚未关闭，则需要将其关闭后再进行升级。具体操作如下所示：
```
Comment (TomShawn) [inline sql-plan-management.md:327]: v5.2 以前的版本也指 v3.1/v3.0/v2.1?
```suggestion
* 当你尝试从 v5.2 以前的版本（不含 v4.0）升级到此版本，需要注意在升级前检查自动演进的开关 `tidb_evolve_plan_baselines` 是否已经关闭。如果尚未关闭，则需要将其关闭后再进行升级。具体操作如下所示：
```

File: sql-plan-management.md [hunk 3]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+
+  ```sql
+  -- 在待升级的版本上检查自动演进的开关 `tidb_evolve_plan_baselines` 是否关闭。
+  
+  select @@global.tidb_evolve_plan_baselines;
+  
+  -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
+  
+  set global tidb_evolve_plan_baselines = off;
+  ```
```
Comment (TomShawn) [inline sql-plan-management.md:339]:
```suggestion
    {{< copyable "sql" >}}

    ```sql
    -- 在待升级的版本上检查自动演进的开关 `tidb_evolve_plan_baselines` 是否关闭。
  
    select @@global.tidb_evolve_plan_baselines;
  
    -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
  
    set global tidb_evolve_plan_baselines = off;
    ```
```

File: sql-plan-management.md [hunk 4]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+
+* 当你尝试从 v4 版本升级到此版本，需要注意在升级前检查所有可用绑定对应的查询语句在新版本中是否存在语法错误。如果存在语法错误，则需要将对应的绑定进行删除。
```
Comment (TomShawn) [inline sql-plan-management.md:341]:
```suggestion
* 当你尝试从 v4.0 版本升级到此版本，需要注意在升级前检查所有可用绑定对应的查询语句在新版本中是否存在语法错误。如果存在语法错误，则需要删除对应的绑定。
```

File: sql-plan-management.md [hunk 5]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+  
+  -- 将上一条查询得到的结果，在新版本的测试环境中进行验证。
+  
+  bind_sql_0;
+  bind_sql_1;
+  ...
+  
+  -- 如果报错信息是语法错误（ERROR 1064 (42000): You have an error in your SQL syntax），则需要删除对应的绑定。
+  -- 如果是其他错误，如未找到表，则表示语法兼容，不需要进行额外的处理。
+  ```
```
Comment (TomShawn) [inline sql-plan-management.md:360]:
```suggestion
  具体操作如下所示：

  {{< copyable "sql" >}}

    ```sql
    -- 在待升级的版本上检查现有可用绑定对应的查询语句。
  
    select bind_sql from mysql.bind_info where source != 'builtin' and status = 'using';
  
    -- 将上一条查询得到的结果，在新版本的测试环境中进行验证。
  
    bind_sql_0;
    bind_sql_1;
    ...
  
    -- 如果报错信息是语法错误（ERROR 1064 (42000): You have an error in your SQL syntax），则需要删除对应的绑定。
    -- 如果是其他错误，如未找到表，则表示语法兼容，不需要进行额外的处理。
    ```
```

File: sql-plan-management.md (hunk 2 above)
Comment (Reminiscent) [reply sql-plan-management.md:327]: 我们现在应该不支持跨大的版本升级，所以这里应该指的是 v4.0/v5.0/v5.1 这三个版本

File: sql-plan-management.md [hunk 6]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+  -- 在待升级的版本上检查自动演进的开关 `tidb_evolve_plan_baselines` 是否关闭。
+  
+  select @@global.tidb_evolve_plan_baselines;
+  
+  -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
+  
+  set global tidb_evolve_plan_baselines = off;
+  ```
+
+* 当你尝试从 v4.0 版本升级到此版本，需要注意在升级前检查所有可用绑定对应的查询语句在新版本中是否存在语法错误。如果存在语法错误，则需要删除对应的绑定。
```
Comment (Reminiscent) [inline sql-plan-management.md:341]: 该检查需要 cherry-pick 到 v5.0 和 v5.1 这两个版本的文档中。

File: sql-plan-management.md [hunk 7]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
     | max_execution_time | 查询过程最多消耗多少时间 |
 
 + `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读。由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
+
+## 升级检查 (Upgrade Checklist)
+
+执行计划管理功能（SPM）在版本升级过程中可能会出现一些兼容性问题导致升级失败，你需要在版本升级前做一些检查，确保版本顺利升级。
+
+* 当你尝试从 v5.2 以前的版本（含 v4.0/ v5.0/ v5.1）升级到此版本，需要注意在升级前检查自动演进的开关 `tidb_evolve_plan_baselines` 是否已经关闭。如果尚未关闭，则需要将其关闭后再进行升级。具体操作如下所示：
```
Comment (TomShawn) [inline sql-plan-management.md:327]:
```suggestion
* 当你尝试从 v5.2 以前的版本（即 v4.0、v5.0、v5.1）升级到此版本，需要注意在升级前检查自动演进的开关 `tidb_evolve_plan_baselines` 是否已经关闭。如果尚未关闭，则需要将其关闭后再进行升级。具体操作如下所示：
```

File: sql-plan-management.md [hunk 8]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+
+  ```sql
+  -- 在待升级的版本上检查自动演进的开关 `tidb_evolve_plan_baselines` 是否关闭。
+  
+  select @@global.tidb_evolve_plan_baselines;
+  
+  -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
+  
+  set global tidb_evolve_plan_baselines = off;
+  ```
```
Comment (TomShawn) [inline sql-plan-management.md:339]:
```suggestion
    ```sql
    -- 在待升级的版本上检查自动演进的开关 `tidb_evolve_plan_baselines` 是否关闭。
  
    select @@global.tidb_evolve_plan_baselines;
  
    -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
  
    set global tidb_evolve_plan_baselines = off;
    ```
```

File: sql-plan-management.md [hunk 9]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
 
 + `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读。由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
+
+## 升级检查 (Upgrade Checklist)
+
+执行计划管理功能（SPM）在版本升级过程中可能会出现一些兼容性问题导致升级失败，你需要在版本升级前做一些检查，确保版本顺利升级。
+
+* 当你尝试从 v5.2 以前的版本（即 v4.0、v5.0、v5.1）升级到此版本，需要注意在升级前检查自动演进的开关 `tidb_evolve_plan_baselines` 是否已经关闭。如果尚未关闭，则需要将其关闭后再进行升级。具体操作如下所示：
+
+  {{< copyable "sql" >}}
```
Comment (TomShawn) [inline sql-plan-management.md:329]:
```suggestion
    {{< copyable "sql" >}}
```

File: sql-plan-management.md [hunk 10]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+    -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
+  
+    set global tidb_evolve_plan_baselines = off;
+    ```
+
+* 当你尝试从 v4.0 版本升级到此版本，需要注意在升级前检查所有可用绑定对应的查询语句在新版本中是否存在语法错误。如果存在语法错误，则需要删除对应的绑定。
+
+  具体操作如下所示：
+
+  {{< copyable "sql" >}}
```
Comment (TomShawn) [inline sql-plan-management.md:345]:
```suggestion
    {{< copyable "sql" >}}
```

File: sql-plan-management.md [hunk 11]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
...
+    select @@global.tidb_evolve_plan_baselines;
+  
+    -- 如果演进的开关 `tidb_evolve_plan_baselines` 尚未关闭，则需要将其关闭。
+  
+    set global tidb_evolve_plan_baselines = off;
+    ```
+
+* 当你尝试从 v4.0 版本升级到此版本，需要注意在升级前检查所有可用绑定对应的查询语句在新版本中是否存在语法错误。如果存在语法错误，则需要删除对应的绑定。
+
+  具体操作如下所示：
```
Comment (TomShawn) [inline sql-plan-management.md:343]:
```suggestion
    具体操作如下所示：
```

File: sql-plan-management.md [hunk 12]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
     | max_execution_time | 查询过程最多消耗多少时间 |
 
 + `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读。由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
+
+## 升级检查 (Upgrade Checklist)
+
+执行计划管理功能（SPM）在版本升级过程中可能会出现一些兼容性问题导致升级失败，你需要在版本升级前做一些检查，确保版本顺利升级。
+
+* 当你尝试从 v5.2 以前的版本（即 v4.0、v5.0、v5.1）升级到此版本，需要注意在升级前检查自动演进的开关 `tidb_evolve_plan_baselines` 是否已经关闭。如果尚未关闭，则需要将其关闭后再进行升级。具体操作如下所示：
```
Comment (TomShawn) [inline sql-plan-management.md:327]: 这里的此版本是指 v5.2 及以后的版本吗？

File: sql-plan-management.md (hunk 6 above)
Comment (TomShawn) [reply sql-plan-management.md:341]: 在 v5.0 和 v5.1 文档中，这个【此版本】是指什么版本呢？

**Review by TomShawn — APPROVED** [review]
LGTM

File: sql-plan-management.md [hunk 13]
```diff
@@ -319,3 +319,43 @@ create global binding for select * from t where a < 100 and b < 100 using select
     | max_execution_time | 查询过程最多消耗多少时间 |
 
 + `read_from_storage` 是一个非常特别的 hint，因为它指定了读表时选择从 TiKV 读还是从 TiFlash 读。由于 TiDB 提供隔离读的功能，当隔离条件变化时，这个 hint 对演进出来的执行计划影响很大，所以当最初创建的绑定中存在这个 hint，TiDB 会无视其所有演进的绑定。
+
+## 升级检查 (Upgrade Checklist)
+
+执行计划管理功能（SPM）在版本升级过程中可能会出现一些兼容性问题导致升级失败，你需要在版本升级前做一些检查，确保版本顺利升级。
```
Comment (yudongusa) [inline sql-plan-management.md:325]: 对外客户文档使用第二人称吗？还是使用第三人称“用户”？前者比较少见。 @TomShawn

Comment (TomShawn) [reply sql-plan-management.md:325]:
```suggestion
执行计划管理功能（SPM）在版本升级过程中可能会出现一些兼容性问题导致升级失败，用户需要在版本升级前做一些检查，确保版本顺利升级。
```

**Comment by yudongusa** [comment]
LGTM

**Review by yudongusa — APPROVED** [review]

**Pull request merged**